metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/sensors/datahub_sensors.py

```diff
@@ -314,100 +315,177 @@ def process_asset_materialization(
+                            "urn:li:dataPlatform:", ""
+                        ),
+                    )
+                    if lineage and lineage.downstreams:
+                        if self.config.emit_queries:
+                            dagster_generator.gen_query_aspect(
+                                graph=self.graph,
+                                platform=asset_downstream_urn.platform,
+                                query_subject_urns=lineage.upstreams
+                                + lineage.downstreams,
+                                query=str(query_metadata.text),
+                            )
+                        downstreams = downstreams.union(set(lineage.downstreams))
+                        upstreams = upstreams.union(set(lineage.upstreams))
+                        context.log.info(
+                            f"Upstreams: {upstreams} Downstreams: {downstreams}"
+                        )
+                    else:
+                        context.log.info(f"Lineage not found for {query_metadata.text}")
+                else:
+                    context.log.info("Query not found in metadata")
+            except Exception as e:
+                context.log.error(f"Error in processing asset logs: {e}")
+                context.log.error(f"Traceback: {traceback.format_exc()}")
+
+        return upstreams, downstreams
+
+    def _emit_or_connect_asset(
+        self,
+        context: RunStatusSensorContext,
+        dagster_generator: DagsterGenerator,
+        log: EventLogEntry,
+        asset_key: List[str],
+        properties: Dict[str, str],
+        upstreams: Set[str],
+        downstreams: Set[str],
+        dataset_inputs: Dict[str, Set[DatasetUrn]],
+        dataset_outputs: Dict[str, Set[DatasetUrn]],
+    ) -> None:
+        if self.config.emit_assets:
+            context.log.info("Emitting asset metadata...")
+            dataset_urn = dagster_generator.emit_asset(
+                self.graph,
+                asset_key,
+                log.asset_materialization.description
+                if log.asset_materialization
+                else None,
+                properties,
+                downstreams=downstreams,
+                upstreams=upstreams,
+                materialize_dependencies=self.config.materialize_dependencies,
+            )
+            if log.step_key:
                 dataset_outputs[log.step_key].add(dataset_urn)
+        else:
+            context.log.info(
+                "Not emitting assets but connecting materialized dataset to DataJobs"
+            )
+            if log.step_key:
+                dataset_outputs[log.step_key] = dataset_outputs[log.step_key].union(
+                    [DatasetUrn.from_string(d) for d in downstreams]
+                )
+                dataset_inputs[log.step_key] = dataset_inputs[log.step_key].union(
+                    [DatasetUrn.from_string(u) for u in upstreams]
```

Comment:
do we need any error handling around invalid urns? or will those get logged elsewhere?